seastats/stats.py

```diff
@@ -220,52 +243,67 @@ def get_stats(
-        version_stat["intercept"] = intercept
-    if should_calculate("slope_pp") or should_calculate("intercept_pp"):
-        slopepp, interceptpp = get_slope_intercept_pp(sim, obs)
-        version_stat["slope_pp"] = slopepp
-        version_stat["intercept_pp"] = interceptpp
-    if should_calculate("mad"):
-        version_stat["mad"] = get_mad(sim, obs, round)
-    if should_calculate("madp"):
-        version_stat["madp"] = get_madp(sim, obs, round)
-    if should_calculate("madc"):
-        version_stat["madc"] = get_madc(sim, obs, round)
-    if should_calculate("kge"):
-        version_stat["kge"] = get_kge(sim, obs, round)
+    for metric in metrics:
+        match metric:
+            case "bias":
+                version_stat["bias"] = get_bias(sim, obs, round)
+            case "rmse":
+                version_stat["rmse"] = get_rmse(sim, obs, round)
+            case "rms":
+                version_stat["rms"] = get_rms(sim, obs, round)
+            case "rms_95":
+                version_stat["rms_95"] = get_rms_quantile(sim, obs, 0.95, round)
+            case "sim_mean":
+                version_stat["sim_mean"] = np.round(sim.mean(), round)
+            case "obs_mean":
+                version_stat["obs_mean"] = np.round(obs.mean(), round)
+            case "sim_std":
+                version_stat["sim_std"] = np.round(sim.std(), round)
+            case "obs_std":
+                version_stat["obs_std"] = np.round(obs.std(), round)
+            case "nse":
+                version_stat["nse"] = get_nse(sim, obs, round)
+            case "lamba":
+                version_stat["lamba"] = get_lambda(sim, obs, round)
+            case "cr":
+                version_stat["cr"] = get_corr(sim, obs, round)
+            case "cr_95":
+                version_stat["cr_95"] = get_corr_quantile(sim, obs, 0.95, round)
+            case "slope":
+                slope, intercept = get_slope_intercept(sim, obs, round)
+                version_stat["slope"] = slope
+                version_stat["intercept"] = intercept
+            case "intercept":
+                if "slope" in metrics:
+                    pass  # because already computed
+                else:
+                    slope, intercept = get_slope_intercept(sim, obs, round)
+                    version_stat["slope"] = slope
+                    version_stat["intercept"] = intercept
+            case "slope_pp":
+                slopepp, interceptpp = get_slope_intercept_pp(sim, obs, round)
+                version_stat["slope_pp"] = slopepp
+                version_stat["intercept_pp"] = interceptpp
+            case "intercept_pp":
+                if "slope_pp" in metrics:
+                    pass  # because already computed
+                else:
+                    slopepp, interceptpp = get_slope_intercept_pp(sim, obs, round)
+                    version_stat["slope_pp"] = slopepp
+                    version_stat["intercept_pp"] = interceptpp
```

Comment:
For the normal slope/intercept, it's okay indeed. 
But for the `*_pp` (percentiles) it does affect runtime. 
That's why I wanted to avoid to calculate it twice.